cpp/src/arrow/extension/fixed_shape_tensor.cc

```diff
@@ -203,6 +215,40 @@ std::shared_ptr<Array> FixedShapeTensorType::MakeArray(
   return std::make_shared<ExtensionArray>(data);
 }
 
+const Result<std::shared_ptr<Tensor>> FixedShapeTensorType::GetTensor(
+    const std::shared_ptr<ExtensionScalar>& scalar) const {
+  const auto array =
+      internal::checked_pointer_cast<const FixedSizeListScalar>(scalar->value)->value;
+  const auto value_type =
+      internal::checked_pointer_cast<FixedWidthType>(this->value_type());
```

Comment:
This should first validate that it is indeed a fixed width type (and if not raise a proper error message) instead of just casting? 
Also, it should probably check that there are no nulls?

(and add tests for this)

Reply:
Added checks and tests.